server.go

```diff
@@ -153,7 +154,18 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, span = tracer.Start(ctx, "ServeHTTP")
 	defer span.End()
 
-	s.PreRequest(w, r)
+	callbackCalled := false
+	s.AroundRequest(w, r, func() {
```

Comment:
>It seems like we're pretty far away from the configRouteEntry from here. Would it make sense to add a PathPattern field to the Route, then only call AroundRequest if we have a matching route?

I'm 👍 to adding a `PathPattern` Field. I think we should always call `AroundRequest` though, even if there is no matching route.

One idea (similar to what was mentioned in Slack) is to set the `route` and `parameters` on `ctx` via `context.WithValue` so each middleware can access them. Thoughts?

Reply:
https://github.com/BlakeWilliams/viewproxy/pull/36/commits/fb389580027e5b887d3a54cae70f593cd975f674 sets the view proxy path in the request context. I like the middleware approach but I think we can do that in a separate PR

Reply:
I agree, and 👍 to deferring the middleware approach.